src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +428,76 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
+        {
+            while (true)
+            {
+                if (multiHandle.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                var count = multiHandle.FdList.Count;
+                if (count == 0)
+                {
+                    await Task.Delay(100).ConfigureAwait(false);
```

Comment:
This is an expensive way to set up such polling.  It'd be better to just use a single Timer.  You could either make it a repeating timer, or you could make it a single-use timer which you reset via Change in order to restart it during the callback (the latter will ensure you don't have multiple callbacks for the timer in flight at the same time).